source/common/upstream/cluster_manager_impl.h

```diff
@@ -44,23 +45,21 @@ namespace Upstream {
  */
 class ProdClusterManagerFactory : public ClusterManagerFactory {
 public:
-  ProdClusterManagerFactory(Server::Admin& admin, Runtime::Loader& runtime, Stats::Store& stats,
-                            ThreadLocal::Instance& tls, Network::DnsResolverSharedPtr dns_resolver,
-                            Ssl::ContextManager& ssl_context_manager,
-                            Event::Dispatcher& main_thread_dispatcher,
-                            const LocalInfo::LocalInfo& local_info,
-                            Secret::SecretManager& secret_manager,
-                            ProtobufMessage::ValidationContext& validation_context, Api::Api& api,
-                            Http::Context& http_context, Grpc::Context& grpc_context,
-                            Router::Context& router_context,
-                            AccessLog::AccessLogManager& log_manager,
-                            Singleton::Manager& singleton_manager, const Server::Options& options)
+  ProdClusterManagerFactory(
+      Server::Admin& admin, Runtime::Loader& runtime, Stats::Store& stats,
+      ThreadLocal::Instance& tls, Network::DnsResolverSharedPtr dns_resolver,
+      Ssl::ContextManager& ssl_context_manager, Event::Dispatcher& main_thread_dispatcher,
+      const LocalInfo::LocalInfo& local_info, Secret::SecretManager& secret_manager,
+      ProtobufMessage::ValidationContext& validation_context, Api::Api& api,
+      Http::Context& http_context, Grpc::Context& grpc_context, Router::Context& router_context,
+      AccessLog::AccessLogManager& log_manager, Singleton::Manager& singleton_manager,
+      const Server::Options& options, Quic::QuicStatNames& quic_stat_names)
```

Comment:
WDYT of replacing the QuicStatNames args here and elsewhere in this PR with a new class Quic::Context. This would parallel the Http::Context and Grpc::Context names above, and would contain (own) the QuicStatNames. This would enable you to easily plumb in other things you might need globally in Quic.

I'm thinking in particular of a new Descriptor object that @RyanTheOptimist might want to add in another PR. If that never happens it's OK, but that way you'd be laying conduit that could be used for other cables in the future.

Thanks!

Reply:
I chatted with Ryan offline. He has no plan to land the CounterGroup change in the near future. So for now I think we can skip introducing the QuicContext wrapper.
If in a few months he decides to work on that change again, I can do a refactor to make the conduit.